pytorch_lightning/loops/fit_loop.py

```diff
@@ -156,8 +137,8 @@ def done(self) -> bool:
         is reached.
         """
         # TODO(@awaelchli): Move track steps inside training loop and move part of these condition inside training loop
-        stop_steps = FitLoop._is_max_limit_enabled(self.max_steps) and self.global_step >= self.max_steps
-        stop_epochs = FitLoop._is_max_limit_enabled(self.max_epochs) and self.current_epoch >= self.max_epochs
+        stop_steps = self.max_steps != -1 and self.global_step >= self.max_steps
```

Comment:
@awaelchli do you think this indirection is justified for such a simple check?

what if the logic changes slightly? or maybe a property now checks `>` instead of `>=`. The utility becomes useless and it's something else to deprecate.

For example, in `checkpoint_connector` we cannot use it.

Reply:
I don't care, you can choose :) 

But I disagree that we need to deprecate these utilities. First of all they are all protected, second they are under `loops.utilities` so clearly not meant for any other purpose than loops.